pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -174,14 +174,8 @@ func (pr *PipelineRun) HasTimedOut(ctx context.Context, c clock.PassiveClock) bo
 
 // GetServiceAccountName returns the service account name for a given
 // PipelineTask if configured, otherwise it returns the PipelineRun's serviceAccountName.
-func (pr *PipelineRun) GetServiceAccountName(pipelineTaskName string) string {
-	serviceAccountName := pr.Spec.ServiceAccountName
-	for _, sa := range pr.Spec.ServiceAccountNames {
-		if sa.TaskName == pipelineTaskName {
-			serviceAccountName = sa.ServiceAccountName
-		}
-	}
-	return serviceAccountName
+func (pr *PipelineRun) GetServiceAccountName() string {
```

Comment:
same as above, this can be removed

Reply:
Ah, my old "don't change anything you don't have to" instinct going overboard... =) On it.